pkg/sysregistriesv2/system_registries_v2_test.go

```diff
@@ -463,3 +463,18 @@ func TestTryUpdatingCache(t *testing.T) {
 	assert.Nil(t, registries)
 	assert.Equal(t, 1, len(configCache))
 }
+
+func TestRegistriesConfDirectory(t *testing.T) {
+	ctx := &types.SystemContext{
+		SystemRegistriesConfPath:    "testdata/base-for-registries.d.conf",
+		SystemRegistriesConfDirPath: "testdata/registries.conf.d",
+	}
+	configCache = make(map[string]*V2RegistriesConf)
+	registries, err := TryUpdatingCache(ctx)
+	assert.Nil(t, err)
+	assert.NotNil(t, registries)
+
+	assert.Equal(t, registries.UnqualifiedSearchRegistries, []string{"example-overwrite.com"})
+	assert.Equal(t, len(registries.Registries), 1)
+	assert.Equal(t, registries.Registries[0].Location, "2.com")
```

Comment:
Why is this correct? Wasn’t the configuration for “`base.com` should be insecure” (as well as the empty configuration for `1.com` completely lost?

Reply:
The `[[registry]]` tables are not "merged" but entirely replaced if specified. Hence, we first replace "base.com" with "1.com" which is finally replaced with "2.com".

Reply:
Sure, I understand that’s what the implementation does. I’m suggesting that this is not what users would expect or consider reasonable.

(It does, I suppose, work for the search path overrides this exists for… but in that case it almost might be safer to just not support `registries` overrides in `*.d` instead of overriding it as a single unit.)

Reply:
We can extend the logic to replace items in the `[[registry]]` table (based on the location). It should be fairly straight forward.

Reply:
That would be ideal.

(I’m not quite sure how this would work for v1, but this is a new directory — so why not make it v2 only, and do the v1→v2 conversion only on the old `.conf` file?)

Reply:
This sounds like a sane approach. I'll spin up a new PR tomorrow morning :+1: 

@mrunalp FYI